src/api/cpp/cvc5.cpp

```diff
@@ -6099,218 +6099,207 @@ Term Solver::mkTuple(const std::vector<Sort>& sorts,
 /* Create operators                                                           */
 /* -------------------------------------------------------------------------- */
 
-Op Solver::mkOp(Kind kind) const
+Op Solver::mkOp(Kind kind, const std::vector<uint32_t>& args) const
 {
   CVC5_API_TRY_CATCH_BEGIN;
   CVC5_API_KIND_CHECK(kind);
-  CVC5_API_CHECK(s_indexed_kinds.find(kind) == s_indexed_kinds.end())
-      << "Expected a kind for a non-indexed operator.";
   //////// all checks before this line
-  return Op(this, kind);
-  ////////
-  CVC5_API_TRY_CATCH_END
-}
+  size_t nargs = args.size();
 
-Op Solver::mkOp(Kind kind, const std::string& arg) const
-{
-  CVC5_API_TRY_CATCH_BEGIN;
-  CVC5_API_KIND_CHECK(kind);
-  CVC5_API_KIND_CHECK_EXPECTED((kind == DIVISIBLE), kind) << "DIVISIBLE";
-  //////// all checks before this line
-  Op res;
-  /* CLN and GMP handle this case differently, CLN interprets it as 0, GMP
-   * throws an std::invalid_argument exception. For consistency, we treat it
-   * as invalid. */
-  CVC5_API_ARG_CHECK_EXPECTED(arg != ".", arg)
-      << "a string representing an integer, real or rational value.";
-  res = Op(this,
-           kind,
-           *mkValHelper<cvc5::Divisible>(cvc5::Divisible(cvc5::Integer(arg)))
-                .d_node);
-  return res;
-  ////////
-  CVC5_API_TRY_CATCH_END;
-}
+  if (nargs == 0)
+  {
+    CVC5_API_CHECK(s_indexed_kinds.find(kind) == s_indexed_kinds.end())
+        << "Expected a kind for a non-indexed operator.";
+    return Op(this, kind);
+  }
 
-Op Solver::mkOp(Kind kind, uint32_t arg) const
-{
-  CVC5_API_TRY_CATCH_BEGIN;
-  CVC5_API_KIND_CHECK(kind);
-  //////// all checks before this line
   Op res;
   switch (kind)
   {
+    // operator kinds with arity 1
     case DIVISIBLE:
+      CVC5_API_OP_CHECK_ARITY(nargs, 1, kind);
       res = Op(this,
                kind,
-               *mkValHelper<cvc5::Divisible>(cvc5::Divisible(arg)).d_node);
+               *mkValHelper<cvc5::Divisible>(cvc5::Divisible(args[0])).d_node);
```

Comment:
I don't like this pattern: we specify the type twice, `mkValHelper` creates an `api::Term` just to extract the node and throw the term away, and the `Op` creation is created over and over again.
How about we use a new helper `mkOpHelper` instead of `mkValHelper` that looks roughly like this:
```
template<typename T>
Op mkOpHelper(Kind kind, const T& t)
{
  Node res = getNodeManager()->mkConst(t);
  static_cast<void>(res.getType(true)); /* kick off type checking */
  return Op(this, kind, res);
}
```
then the above would become
```
CVC5_API_OP_CHECK_ARITY(nargs, 1, kind);
res = mkOpHelper(kind, cvc5::Divisible(args[0]));
```

Reply:
Also, we could immediately return the `Op` instead of storing it in `res`, moving the `!res.isNul()` check into the new helper.

Reply:
Good point, I'll fix this.